sensor/common/compliance/nodescan_handler_impl.go

```diff
@@ -0,0 +1,82 @@
+package compliance
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+)
+
+type nodeScanHandlerImpl struct {
+	nodeScans <-chan *storage.NodeScanV2
+	toCentral chan *central.MsgFromSensor
+
+	stopC    concurrency.ErrorSignal
+	stoppedC concurrency.ErrorSignal
+}
+
+func (c *nodeScanHandlerImpl) Capabilities() []centralsensor.SensorCapability {
+	return []centralsensor.SensorCapability{centralsensor.NodeScanningCap}
+}
+
+func (c *nodeScanHandlerImpl) ResponsesC() <-chan *central.MsgFromSensor {
+	return c.toCentral
+}
+
+func (c *nodeScanHandlerImpl) Start() error {
+	// c.run closes chan toCentral on exit, so we do not want to close twice in case of a restart
+	if !c.stopC.IsDone() {
+		go c.run()
+		return nil
+	}
+	return errors.New("stopped handlers cannot be restarted")
+}
+
+func (c *nodeScanHandlerImpl) Stop(err error) {
+	c.stopC.SignalWithError(err)
+}
+
+func (c *nodeScanHandlerImpl) ProcessMessage(_ *central.MsgToSensor) error {
+	// This component doesn't actually process or handle any messages sent from Central to Sensor (yet).
```

Comment:
Does this imply there is a plan to add this (I'm imagining Central requesting a scan periodically as part of reprocessing or user forcing it)?

Reply:
This is added to implement the handler interface. It does not meant that we will support anything arriving from central, but we do have such possibility of we want to :)